Pokemon Unity/Assets/Scripts2/Saving/SaveData.cs

```diff
@@ -66,49 +86,71 @@ public void AddPokemonCenter(int scene, SerializableVector3 position, int direct
             pCenterFdirection = fDirection;
         }
 
-        public SaveData(
+        public SaveData
+            (
             string saveName,
-            int activeScene,
-            string playerName, int trainerID, int secretID, bool isMale,
-            //List<GymBadges> gymBadges,
+
+            Settings.Languages language,
+            byte windowBorder,
+            byte dialogBorder,
+            byte textSpeed,
+            float mvol,
+            float svol,
+
+            string playerName,
+            int trainerID,
+            int secretID,
+            bool isMale,
+
             bool?[] pokedex,
-            System.TimeSpan playerTime,
-            SerializableVector3 playerPosition, int playerDirection,
-            SerializableVector3 followerPosition, int followerDirection,
-            Pokemon.Pokemon[] playerParty, Pokemon.Pokemon[,] playerPC, List<Items> playerBag,
+            TimeSpan playerTime,
+
+            SerializableVector3 playerPosition,
+            int playerDirection,
+            SerializableVector3 followerPosition,
+            int followerDirection,
+
+            int activeScene,
+
+            Pokemon.Pokemon[] playerParty,
+            SeriPC pc,
+            List<Items> playerBag,
+
             List<SaveEvent> eventList
             )
         {
-            //Important
             SaveName = saveName;
-            ActiveScene = activeScene;
 
-            //Player Information
+            Language = (int)language;
+            WindowBorder = windowBorder;
+            DialogBorder = dialogBorder;
+            TextSpeed = textSpeed;
+            mVol = mvol;
+            sVol = svol;
+
             PlayerName = playerName;
-            playerID = (trainerID + secretID) * 65536;
```

Comment:
That was actually the math they used in the trainer assembly from nintendo. not sure on all of the details, but the idea is longer than 5 characters. it's an entire hash-code, which technically SHOULD be rewritten, i wouldnt technically delete the information until it's resolved. Maybe comment it out instead? I personally would move the `playerId` logic to `override GetHasCode`, and substitute the field with a property and only assign it a `get` (since the set is automatically done by engine).

Reply:
Oh! just realized this was your save data class. My bad, thought it was on the original source class